spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -17,121 +17,493 @@
+            }
+        }
+        else {
+            product = ownerProductApi.createProductByOwner(owner.getKey(), product);
+            product30 = ownerProductApi.createProductByOwner(owner.getKey(), product30);
+            content = ownerContentApi.createContent(owner.getKey(), content);
+            archContent = ownerContentApi.createContent(owner.getKey(), archContent);
+            product = ownerProductApi.addContent(owner.getKey(), product.getId(), content.getId(), false);
+            product = ownerProductApi.addContent(owner.getKey(), product.getId(), archContent.getId(), false);
+            pool = Pools.random(product)
+                .contractNumber("12345")
+                .accountNumber("6789")
+                .orderNumber("order1");
+            pool = ownerApi.createPool(owner.getKey(), pool);
+            pool30 = Pools.random(product30)
+                .contractNumber("123456")
+                .accountNumber("67890")
+                .orderNumber("order2");
+            pool30 = ownerApi.createPool(owner.getKey(), pool30);
+        }
+        system = Consumers.random(owner)
+            .name(StringUtil.random("system"))
+            .type(new ConsumerTypeDTO().label("system"))
+            .facts(Map.of("system.certificate_version", "3.4", "uname.machine", "i386"));
+        system = consumerApi.register(system);
+    }
+
+    /**
+     * This test covers the case where the system supports 3.0 certs, but
+     * the server is creating 3.4 certs, and the product contains attributes
+     * supported by 3.0.
+     * @throws Exception
+     */
+    @Test
+    @DisplayName("generated a version 3.4 certificate when requesting a 3.0 certificate")
+    public void shouldGenerateThreeFourCertRequestThreeZero() throws Exception {
+        ConsumerDTO v3System = Consumers.random(owner)
+            .name(StringUtil.random("system"))
+            .type(new ConsumerTypeDTO().label("system"))
+            .facts(Map.of("system.certificate_version", "3.0", "uname.machine", "i386"));
+        v3System = consumerApi.register(v3System);
+        consumerApi.bindProduct(v3System.getUuid(), product30.getId());
+        String value = CertificateUtil.standardExtensionValueFromCert(
+            consumerApi.listEntitlements(v3System.getUuid())
+            .get(0).getCertificates().iterator().next().getCert(), "1.3.6.1.4.1.2312.9.6");
+        assertEquals("3.4", value);
+    }
+
+    @Test
+    @DisplayName("generated a version 3.4 certificate")
+    public void shouldGenerateThreeFourCert() throws Exception {
+        consumerApi.bindProduct(system.getUuid(), product.getId());
+        EntitlementDTO ent = consumerApi.listEntitlements(system.getUuid()).get(0);
+        String value = CertificateUtil.standardExtensionValueFromCert(
+            ent.getCertificates().iterator().next().getCert(), "1.3.6.1.4.1.2312.9.6");
+        assertEquals("3.4", value);
+        consumerApi.unbindAll(system.getUuid());
```

Comment:
If the `system` is created in the `beforeAll`, why do you need to unbind from it at the end of each test? It should be a separate consumer for each one

Reply:
Carry over from the ruby